### PR TITLE
Update retry handler's to retry on gRPC error `INTERNAL` only if it matches `RST_STREAM` regex

### PR DESCRIPTION
**Pull request by danielakhterov**
Resolves: #430 

Warning: This is untested since all the nodes seem to work fine right now.

### CODE REVIEW

File: src/topic/TopicMessageQuery.js [hunk 1]
```diff
@@ -115,16 +116,18 @@ export default class TopicMessageQuery {
...
                     );
                 } else {
                     switch (error.code) {
-                        // NOT_FOUND
-                        // eslint-disable-next-line no-fallthrough
-                        case 5:
                         // RESOURCE_EXHAUSTED
                         // eslint-disable-next-line no-fallthrough
                         case 8:
+                            return RST_STREAM.test(error.details.toString());
```
Comment (steven-sheehy) [inline src/topic/TopicMessageQuery.js:127]: This should be for case 13 INTERNAL

File: src/topic/TopicMessageQuery.js [hunk 2]
```diff
@@ -115,16 +116,18 @@ export default class TopicMessageQuery {
                         error.toString().includes("NOT_FOUND") ||
                         error.toString().includes("UNAVAILABLE") ||
                         error.toString().includes("RESOURCE_EXHAUSTED") ||
-                        error.toString().includes("INTERNAL")
+                        (error.toString().includes("INTERNAL") &&
```
Comment (steven-sheehy) [inline src/topic/TopicMessageQuery.js:119]: Why would there be a non-gRPC `StatusObject` that includes gRPC status codes? It seems like errors in this if statement would more likely to be lower level HTTP2 errors, right? In which case you might want to just retry all if so.

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

**Comment by steven-sheehy** [comment]
> Warning: This is untested since all the nodes seem to work fine right now.

I'm not sure how you can say that. Testnet node 0.0.6 has this issue right now. Your CI workflow is failing right now with this exact error. It seems like that regex is not enough to catch the proxy issue. You can probably make it more generic and just search for `\brst[^0-9a-zA-Z]stream\b`.

File: src/topic/TopicMessageQuery.js (hunk 2 above)
Comment (mehcode) [reply src/topic/TopicMessageQuery.js:119]: This really isn't part of this PR but I do agree that this whole part of the default retry handler should likely be thrown out and we should test for a "gRPC error" more reliably somehow.

File: src/Executable.js [hunk 3]
```diff
@@ -16,6 +16,8 @@ export const ExecutionState = {
     Error: "Error",
 };
 
+export const RST_STREAM = new RegExp(".*RST_STREAM.*");
```
Comment (mehcode) [inline src/Executable.js:19]: You want to use literals here - `/foo/u` - in JS.

File: src/Executable.js [hunk 4]
```diff
@@ -193,7 +195,8 @@ export default class Executable {
         return (
             error.status === GrpcStatus.Unavailable ||
             error.status === GrpcStatus.ResourceExhausted ||
-            error.status === GrpcStatus.Internal
+            (error.status === GrpcStatus.Internal &&
+                RST_STREAM.test(error.message))
```
Comment (steven-sheehy) [inline src/Executable.js:199]: Should have a unit test that at least tests the retry handler with various errors to ensure it produces the desired boolean.

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: src/topic/TopicMessageQuery.js (hunk 2 above)
Comment (danielakhterov) [reply src/topic/TopicMessageQuery.js]: Not entirely sure on what you're suggesting. I did remove that block of code and just `return true`, but I could be misunderstanding what you want me to do here.

**Review by steven-sheehy — APPROVED** [review]

**Pull request merged**